File: frontend/src/app/dto/animal/AnimalResponse.ts
import { AnimalCategory } from './AnimalCategory';
import { AnimalStatus } from './AnimalStatus';
import { Gender } from './Gender';

export interface StatusHistoryEntry {
  status: AnimalStatus;
  changed_at: string;
  changed_by: string;
  changed_by_username: string;
  note: string | null;
}

export interface AnimalResponse {
  id: string;
  name: string;
  category: AnimalCategory;
  breed: string | null;
  gender: Gender;
  age_months: number | null;
  weight: number | null;
  color: string | null;
  chip_id: string | null;
  status: AnimalStatus;
  description: string | null;
  image_paths: string[];
  status_history: StatusHistoryEntry[];
  registered_by: string;
  registered_by_username: string;
  created_at: string;
  updated_at: string;
}

export interface AnimalListResponse {
  success: boolean;
  animals: AnimalResponse[];
  total: number;
}

export interface ImageUploadResponse {
  success: boolean;
  image_paths: string[];
  message: string;
}

export interface MessageResponse {
  success: boolean;
  message: string;
}

File: backend/animal-registry/src/main/java/com/animalshelter/animalregistry/dto/AnimalResponse.java
package com.animalshelter.animalregistry.dto;

import com.animalshelter.animalregistry.model.*;

import java.time.LocalDateTime;
import java.util.List;

public record AnimalResponse(
        String id,
        String name,
        AnimalCategory category,
        String breed,
        Gender gender,
        Integer ageMonths,
        Double weight,
        String color,
        String chipId,
        AnimalStatus status,
        String description,
        List<String> imagePaths,
        List<StatusHistoryEntry> statusHistory,
        String registeredBy,
        String registeredByUsername,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
    public static AnimalResponse fromEntity(Animal animal) {
        return new AnimalResponse(
                animal.getId(),
                animal.getName(),
                animal.getCategory(),
                animal.getBreed(),
                animal.getGender(),
                animal.getAgeMonths(),
                animal.getWeight(),
                animal.getColor(),
                animal.getChipId(),
                animal.getStatus(),
                animal.getDescription(),
                animal.getImagePaths(),
                animal.getStatusHistory(),
                animal.getRegisteredBy(),
                animal.getRegisteredByUsername(),
                animal.getCreatedAt(),
                animal.getUpdatedAt()
        );
    }
}
